plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -327,6 +329,6 @@ func (s *SQLServer) accRow(query Query, acc telegraf.Accumulator, row scanner) e
 
 func init() {
 	inputs.Add("sqlserver", func() telegraf.Input {
-		return &SQLServer{}
+		return NewSQLServer()
```

Comment:
Why not just return `&SQLServer{Servers: []string{defaultServer}` here? This would be a minimal change. Why do you need this new function?

Reply:
@srebhan - This was introduced with the thought that if we need extra processing steps before creating the initializing object, then those can be done in this function. I see such usage in other plug-ins too. Having said that, I see no harm in directly initializing in the init method instead of creating a separate method. Do you see overhead with an extra function though?

Reply:
I don't like introducing machinery "for later use" as too often I have seen that such uses don't arise. :-) However, I also acknowledge that it's a matter of taste and as such it's ok for me if you like to keep it. But if you ask me I would directly initialize. :-)
Please let me know if you want to keep the function!

Reply:
@srebhan - Thank you for your thoughts on this. I'll go with your point of minimal changes and leaving the future-use to future :) I'll push the relevant changes and ping back.

Reply:
@srebhan - Updated as per above conversation